charts/merlin/values.yaml

```diff
@@ -26,167 +26,241 @@ deployment:
+        AlertBranch: master
+      WardenConfig:
+        APIHost: ""
+  ReactAppConfig:
+    DocURL:
+      - Label: "Getting Started with Merlin"
+        Href: "https://github.com/gojek/merlin/blob/main/docs/getting-started/README.md"
+    # -- Comma-separated value of Docker registries that can be chosen in deployment page
+    DockerRegistries: ghcr.io/gojek,ghcr.io/your-company
+    Environment: dev
+    FeastCoreURL: http://feast-core.mlp:8080/v1
+    HomePage: /merlin
+    MaxAllowedReplica: 20
+    MerlinURL: /api/merlin/v1
+    MlpURL: /api
+    OauthClientID:  # to be set via CICD pipeline
+    UPIDocumentation: "https://github.com/caraml-dev/universal-prediction-interface/blob/main/docs/api_markdown/caraml/upi/v1/index.md"
+    CPUCost:  # Unused
+    MemoryCost:  # Unused
+  StandardTransformerConfig:
+    ImageName: merlin-transformer:1.0.0
+    SimulationFeast:
+      FeastRedisURL: online-serving-redis.feast.dev
+      FeastBigtableURL: online-serving-bt.feast.dev
+    FeastServingURLs:
+      - Host: "online-serving-redis.feast.dev"
+        Label: "Online Serving with Redis"
+        Icon: "redis"
+        SourceType: "REDIS"
+      - Host: "online-serving-bigtable.feast.dev"
+        Label: "Online Serving with BigTable"
+        Icon: "bigtable"
+        SourceType: "BIGTABLE"
+    DefaultServingURL: online-serving-redis.feast.dev
+    ## Redis storage configuration for feast retrieval
+    # FeastRedisConfig:
+    #   IsRedisCluster: true
+    #   IsUsingDirectStorage: false
+    #   ServingURL: "online-serving-redis.feast.dev"
+    #   RedisAddresses:
+    #     - 10.1.1.2
+    #     - 10.1.1.3
+    #   PoolSize: 5
+    #   MaxRetries: 0
+    #   ReadTimeout: "1s"
+    #   WriteTimeout: "1s"
+    #   IdleTimeout: "15s"
+    #   IdleCheckFrequency: "5s"
+    #   MinIdleConn: 2
+    FeastBigtableConfig:
+      ServingURL: "online-serving-bigtable.feast.dev"
+      Project: gcp-project
+      Instance: instance
+      PoolSize: 5
+    ## Bigtable storage configuration for feast retrieval
+    #   IsUsingDirectStorage: false
+    #   AppProfile: default
+    #   KeepAliveInterval: 2m
+    #   KeepAliveTimeout: 1m
+    BigtableCredential:
+    FeastServingKeepAlive:
+      Enabled: false
+      Time: 60s
+      Timeout: 5s
+    DefaultFeastSource: 2
+    FeastCoreURL: core.feast.dev
+    FeastCoreAuthAudience: core.feast.dev
+    EnableAuth: false
+    Jaeger:
+      AgentHost: localhost
+      AgentPort: 6831
+      SamplerType: const
+      SamplerParam: 1
+      Disabled: false
+    Kafka:
+      Brokers: "kafka-brokers"
+      MaxMessageSizeBytes: "1048588"
+  PyfuncGRPCOptions: "{}"
+
+clusterConfig:
+  # -- (bool) Configuration to tell Merlin API how it should authenticate with deployment k8s cluster
+  # By default, Merlin API expects to use a remote k8s cluster for deployment and to do so, it requires
+  # cluster access configurations to be configured as part of values.yaml
+  useInClusterConfig: false
```

Comment:
Ah good catch; initially I added the `useInClusterConfig` field because it was just a copy from the Turing configs but I guess there's no harm adding the extra resources for Merlin too.

Reply:
On second thought, after starting to refactor to the Merlin API for the in-cluster scenario, I realised that we probably need more extensive refactoring to change the way deployment-related configs are currently stored. 

As of present, Merlin stores deployment-related configs such as MaxCPU or MaxMemory are stored in cluster-specific [structs](https://github.com/caraml-dev/merlin/blob/a39436dc55b9c19a9447fe707f5610fcc7c0ef03/api/config/environment.go#L29). Turing, however, stores it in app-level [structs](https://github.com/caraml-dev/turing/blob/11c58f75f00762555813998f4216fd1dc4f0c678/api/turing/config/config.go#L227), which makes these deployment-related configs shared across all clusters it's configured to work with, including the in-cluster scenario, where it simply uses the app-level deployment-configs. 

Since Merlin doesn't have such similar app-level deployment-related configs, and we'd be expecting these cluster-specific configs to be absent if no cluster configs are specified (in-cluster scenario), we'd need to find another way to have some 'default deployment-configs' that the Merlin API can use. We can probably do something similar, and create a new field in the Merlin API configs to store these _default_ deployment-related configs (for the in-cluster scenario). What do you think?

Reply:
Okay, scratch what I just said, after digging a little further, I think even with the `InClusterConfig` flag set as true, even Turing expects at least 1 cluster to be specified in its [environment configs](https://pkg.go.dev/github.com/caraml-dev/turing/api/turing/config#ClusterConfig.EnvironmentConfigs), even if it's the cluster in which the API server is deployed in. The only difference would be the absence of any cluster credentials.

That said, if we're also expecting at least 1 cluster to be specified in the Merlin configs, that cluster will be the cluster that the Merlin API server is deployed in and we would've have the problem mentioned above.

Reply:
Yes, the presence or the absence of the cluster credentials would make the difference. Nonetheless it looks like we need some thought and potentially some app-level changes to support this properly in Merlin. In this case, let's park this problem for now and revisit it when we refactor the environments into MLP. wdyt?

Reply:
Okay I'll abandon all the in-cluster related refactoring on the Merlin API if I can't get it to work. I'll just keep the creation of the new `ClusterRole` and `ClusterRoleBinding` resources then, since they don't really affect anything.